### PR TITLE
improve: optimize build index

### PR DESCRIPTION
**Pull request by Subsegment**
# Required checklist
- [ ] Sample config files updated (`config`,`meta/config` and `default config`) 
- [ ] If there are user-facing changes, the documentation needs to be updated prior to approving the PR( [Link]() )
- [ ] If there are any breaking changes to public APIs, please add the `api change` label.
- [ ] Signed [CLA](https://cla-assistant.io/cnosdb/cnosdb) (if not already signed)

# Which issue does this PR close?

[//]: # (We generally require a GitHub issue to be filed for all bug fixes and enhancements and this helps us generate change logs for our releases. You can link an issue to this PR using the GitHub syntax. For -- example `Closes #123` indicates that this PR will close issue #123.)

Related #.

# Rationale for this change

[//]: # (Why are you proposing this change? If this is already explained clearly in the issue then this section is not needed.
 Explaining clearly why changes are proposed helps reviewers understand your changes and offer better suggestions for fixes.)

# Are there any user-facing changes?

[//]: # (There is no need to duplicate the description in the issue here, but it is sometimes worth providing a summary of the individual changes in this PR.)

### CODE REVIEW

File: tskv/src/index/ts_index.rs [hunk 1]
```diff
@@ -48,16 +54,21 @@ impl TSIndex {
             None => 0,
         };
 
+        let (binlog_change_sender, binlog_change_reciver) = unbounded_channel();
```
Comment (bartliu827) [inline tskv/src/index/ts_index.rs:57]: why use a channel.
why not directly read the file?

File: tskv/src/index/ts_index.rs [hunk 2]
```diff
@@ -577,8 +585,115 @@ pub fn encode_series_id_list(list: &[u32]) -> Vec<u8> {
...
+                            {
+                                let mut storage_w;
+                                loop {
+                                    match ts_index.storage.try_write() {
+                                        Ok(stroage) => {
+                                            storage_w = stroage;
+                                            break;
+                                        }
+                                        Err(_) => {
+                                            tokio::time::sleep(Duration::from_secs(5)).await;
```
Comment (bartliu827) [inline tskv/src/index/ts_index.rs:646]: print notice log

**Comment by ZuoTiJia** [comment]
begin benchmark

**Comment by roseboy-liu** [comment]
https://cnosdb-admin.teleport.cnosdb.com/benchmark-hits/hits_c9362bf.html
https://cnosdb-admin.teleport.cnosdb.com/benchmark-tsbs/tsbs_c9362bf.html

File: tskv/src/index/ts_index.rs [hunk 3]
```diff
@@ -170,45 +181,42 @@ impl TSIndex {
...
-            if let Some(val) = storage_w.get(&key_buf)? {
-                return Ok(byte_utils::decode_be_u32(&val));
+    pub async fn add_series_if_not_exists(
+        &self,
+        series_keys: Vec<SeriesKey>,
+    ) -> IndexResult<Vec<u32>> {
+        let mut ids = Vec::with_capacity(series_keys.len());
+        let mut blocks_data = Vec::new();
+        for mut series_key in series_keys.into_iter() {
+            let id = self.incr_id.fetch_add(1, Ordering::Relaxed) + 1;
```
Comment (bartliu827) [inline tskv/src/index/ts_index.rs:191]: why  add   for  every  key？

**Pull request merged**